k8s/migration/internal/controller/migrationplan_controller.go

```diff
@@ -1497,3 +1509,71 @@ func (r *MigrationPlanReconciler) migrateRDMdisks(ctx context.Context, migration
 	}
 	return nil
 }
+
+// validates that the VM has a valid OS type
+func (r *MigrationPlanReconciler) validateVMOS(vmMachine *vjailbreakv1alpha1.VMwareMachine) (bool, bool, error) {
+	validOSTypes := []string{"windowsGuest", "linuxGuest"}
+	osFamily := strings.TrimSpace(vmMachine.Spec.VMInfo.OSFamily)
+
+	if osFamily == "" || osFamily == "unknown" {
+		r.ctxlog.Info("VM has unknown or unspecified OS type and will be skipped",
+			"vmName", vmMachine.Spec.VMInfo.Name)
+		return false, true, nil
+	}
+
+	for _, validOS := range validOSTypes {
+		if osFamily == validOS {
+			return true, false, nil
+		}
+	}
+
+	return false, false, fmt.Errorf("vm '%s' has an unsupported OS type: %s",
+		vmMachine.Spec.VMInfo.Name, osFamily)
+}
+
+// validates all VMs in the migration plan
+func (r *MigrationPlanReconciler) validateMigrationPlanVMs(
+	ctx context.Context,
+	migrationplan *vjailbreakv1alpha1.MigrationPlan,
+	migrationtemplate *vjailbreakv1alpha1.MigrationTemplate,
+	vmwcreds *vjailbreakv1alpha1.VMwareCreds) ([]*vjailbreakv1alpha1.VMwareMachine, error) {
+	var (
+		validVMs   []*vjailbreakv1alpha1.VMwareMachine
+		skippedVMs []string
+	)
+
+	for _, vmGroup := range migrationplan.Spec.VirtualMachines {
+		for _, vm := range vmGroup {
+			vmMachine, err := GetVMwareMachineForVM(ctx, r, vm, migrationtemplate, vmwcreds)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get VMwareMachine for VM %s: %w", vm, err)
+			}
+
+			valid, skipped, err := r.validateVMOS(vmMachine)
```

Comment:
isn't every skipped VM NOT valid?
ideally we should just need one variable here.